### ISSUE TITLE
Support bulk `external_id` routes for Organizations

### ISSUE THREAD

**Issue by onecrayon** [facetoe/zenpy#146]
Currently to bulk fetch or delete Organizations you have use their Zendesk IDs. However, there are routes for using the `external_id` to handle these tasks, which is preferable when doing something like mirroring data from a site/app/whatever to Zendesk.

It would be lovely if we could do things like the following:

```python
# Assume local_orgs is a list containing local organization objects
orgs = zenpy_client.organizations(external_ids=[i.id for i in local_orgs])
zenpy_client.organizations.delete([Organization(external_id=i.id) for i in local_orgs])
```

I started implementing support for this here:

https://github.com/facetoe/zenpy/compare/master...onecrayon:query-external-ids

But have not had a chance to test it yet (so wasn't comfortable submitting a PR). I will hopefully have a chance to test this out once I'm off work, but would love to hear if you have a better idea for how to implement support for this. :)

**Comment by facetoe** [comment]
I can see how this would be very useful, but we need to be careful with the implementation to ensure that people don't accidentally delete the wrong objects. Adding support for external_ids introduces some ambiguity in how an object is identified, which could in the worse case lead to people deleting the wrong objects accidentally, which would be bad. 

I wonder if it would be better to add methods that explicitly operate on external_id for those endpoints that support them. That way there is no ambiguity. 

What do you think?

**Comment by onecrayon** [comment]
That makes a lot of sense to me. Aside from being wholly untested, the other reason my approach in the branch linked above made me a little uncomfortable was that if you had a mixed set of objects, some with external IDs and some without, you'd end up with a guaranteed error regardless of object order (since the resulting API call would be something like `?external_ids=1,None,3`).

So I guess this would involve two changes:

1. Allow `Organization.external()` to accept a list or a single value
2. Add `Organization.delete_external(api_objects)` (`delete_by_external`?) that deletes by `external_id` instead of `id`

**Comment by facetoe** [comment]
Yep, sounds good. I would add one method though `update_by_external_id` as Zendesk also supports updating objects by external id.

I'll have a look at this when I get the chance, however PRs are welcome :)

**Comment by facetoe** [comment]
Hi @onecrayon, I've implemented this functionality in 8a5ae5cf09cda841dfb4b081fd85f4ee7dcca685 and pushed a new version to PyPy. Let me know if you have any trouble!

**Comment by onecrayon** [comment]
Thanks, @facetoe! I will definitely give this a try and let you know if I run into any issues.
